lib/ace/mode/markdown_worker.js

```diff
@@ -0,0 +1,71 @@
+define(function(require, exports, module) {
+"use strict";
+
+var oop = require("../lib/oop");
+var Mirror = require("../worker/mirror").Mirror;
+var Tokenizer = require("../tokenizer").Tokenizer;
+var BackgroundTokenizer = require("../background_tokenizer").BackgroundTokenizer;
+
+var TextParser = require("./text_parser").TextParser;
+var MarkdownHighlightRules = require("./markdown_highlight_rules").MarkdownHighlightRules;
+
+
+var WORKER_TIMEOUT_MS = 20;
+
+
+var makeSet = function(array) {
+    var tmp = {};
+    for (var i = 0; i < array.length; i++) {
+        tmp[array[i]] = true;
+    }
+    return tmp;
+};
+
+
+var MarkdownWorker = exports.MarkdownWorker = function(sender) {
+    Mirror.call(this, sender);
+    this.setTimeout(WORKER_TIMEOUT_MS);
+    this.mySpellingCheckOptions = {};
+    this.myTypos = {};
+    var highlighter = new MarkdownHighlightRules();
+    this.myBgTokenizer = new BackgroundTokenizer(new Tokenizer(highlighter.getRules()));
+    this.myBgTokenizer.setDocument(this.doc);
+    this.myBgTokenizer.start(0);
+    this.myParser = new TextParser(this.myBgTokenizer);
+};
+
+oop.inherits(MarkdownWorker, Mirror);
```

Comment:
maybe we can extract common code into PapeeriaWorker and extends both LaTeX and Markdown from that worker?